File: core/src/main/java/com/graphhopper/routing/util/parsers/MountainBikeAccessParser.java
package com.graphhopper.routing.util.parsers;

import com.graphhopper.routing.ev.BooleanEncodedValue;
import com.graphhopper.routing.ev.EncodedValueLookup;
import com.graphhopper.routing.ev.Roundabout;
import com.graphhopper.routing.ev.VehicleAccess;
import com.graphhopper.util.PMap;

public class MountainBikeAccessParser extends BikeCommonAccessParser {

    public MountainBikeAccessParser(EncodedValueLookup lookup, PMap properties) {
        this(lookup.getBooleanEncodedValue(VehicleAccess.key("mtb")),
                lookup.getBooleanEncodedValue(Roundabout.KEY));
        blockPrivate(properties.getBool("block_private", true));
        blockFords(properties.getBool("block_fords", false));
    }

    protected MountainBikeAccessParser(BooleanEncodedValue accessEnc, BooleanEncodedValue roundaboutEnc) {
        super(accessEnc, roundaboutEnc);
    }

}

File: core/src/main/kotlin/com/graphhopper/routing/util/parsers/MountainBikeAccessParser.kt
package com.graphhopper.routing.util.parsers

import com.graphhopper.routing.ev.BooleanEncodedValue
import com.graphhopper.routing.ev.EncodedValueLookup
import com.graphhopper.routing.ev.Roundabout
import com.graphhopper.routing.ev.VehicleAccess
import com.graphhopper.util.PMap

open class MountainBikeAccessParser protected constructor(accessEnc: BooleanEncodedValue, roundaboutEnc: BooleanEncodedValue) :
        BikeCommonAccessParser(accessEnc, roundaboutEnc) {

    constructor(lookup: EncodedValueLookup, properties: PMap) : this(
            lookup.getBooleanEncodedValue(VehicleAccess.key("mtb")),
            lookup.getBooleanEncodedValue(Roundabout.KEY)) {
        blockPrivate(properties.getBool("block_private", true))
    }
}
